.github/workflows/main.yml

```diff
@@ -11,9 +11,48 @@ env:
   SOLANA_VERSION: 1.18.12
 
 jobs:
+  format_and_lint:
+    name: Format & lint
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        library: [programs, client-rust, client-js]
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v4
+
+      - name: Setup environment
+        uses: ./.github/actions/setup
+        with:
+          clippy: true
+          node: ${{ env.NODE_VERSION }}
+          rustfmt: true
+          protoc: true
+
+      - name: Format ${{ matrix.library }}
+        run: |
+          if [ "${{ matrix.library }}" == "programs" ]; then
+            pnpm programs:format -- --check
```

Comment:
The `*:format` and `*:lint` script should follow a consistent strategy. In JavaScript, it defaults to checking and I think it should do for Rust as well so we don't need this extra parameter in CI. Again, I think only the user should be "allowed" to offer parameters to these scripts.

Reply:
Okay, this sounds good to me. So we're going to tuck all of the `--check` flags into the scripts, making the scripts all check by default, and it's up to the end user to actually handle the fixes.

If they so choose to add a `--fix` or remove `--check` to the command, they can roll their own script themselves or do this manually at the command line. What do you think?

Reply:
Resolved with:

![CleanShot 2024-07-18 at 18 21 12@2x](https://github.com/user-attachments/assets/0eb2e8d8-b59f-4ac3-bc02-960ed46261b9)